src/core/Akka/Actor/ActorCell.ReceiveTimeout.cs

```diff
@@ -47,7 +47,7 @@ public void SetReceiveTimeout(TimeSpan? timeout=null)
         public void CheckReceiveTimeout()
         {
             CancelReceiveTimeout();
-            if (_receiveTimeoutDuration != null && !Mailbox.HasMessages)
+            if (_receiveTimeoutDuration != null)
```

Comment:
There were some performance concerns about this re-scheduling always now, but the PR was approved anyway.

Reply:
> I agree that this is the most straight forward way to implement it. I'm somewhat concerned about the performance implication of always rescheduling, but other solutions I can think of are too complicated and have other drawbacks.

I agree with what Patrik said.